doc/sphinx/make-your-own-compositor.md

```diff
@@ -0,0 +1,253 @@
+---
+discourse: 4911,5164,5603,6756,8037
+---
+
+# Using Mir Libraries to Build Your Own Wayland Compositor
+If you've seen [the demo tutorial](explore-mir-using-demos), you're probably
+taken away by Mir's capabilities and want to use it and the supporting
+libraries to build your own Wayland compositor. This section will guide you
+through the installation and basic usage of Mir. You'll be even more impressed
+by how easy it is to create a Wayland compositor using Mir!
+
+### A barebones Mir compositor
+
+The following code block is the bare minimum you need to run a Mir compositor:
+```cpp
+#include <miral/runner.h>
+#include <miral/minimal_window_manager.h>
+#include <miral/set_window_management_policy.h>
+
+using namespace miral;
+
+int main(int argc, char const* argv[])
+{
+    MirRunner runner{argc, argv};
+
+    return runner.run_with(
+        {
+            set_window_management_policy<MinimalWindowManager>()
+        });
+}
+```
+
+You can se it's barely longer than the usual C++ "Hello, world!" example. This
+code should give you a minimal compositor with basic window management
+capabilities such as controlling multiple windows, minimizing and maximizing,
+handling input, etc... This is done with the help of MirAL (Mir Abstraction
+Layer) which gives you a high level interface to work with Mir!
+
+To compile, you'll need to install `libmiral`, and
+`mir-graphics-drivers-desktop` which is can be done on different distros as
+follows:
+
+<details>
+<summary> Installing Dependencies on Debian and its derivatives </summary>
+
+```sh
+sudo apt install libmiral-dev mir-graphics-drivers-desktop
+```
+</details>
```

Comment:
Again, I don't think this "click to read on" experience is good.

Reply:
Having Ubuntu as the default and other distros in a collapsible section is a bit better, but I think it still could be better. No idea what can be changed though, perhaps the styling of the collapsible header itself?

Reply:
Agreed. But unless there's a better suggestion I think this is good enough